yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -0,0 +1,418 @@
+				continue
+			}
+			ongoingCmds = append(ongoingCmds, match)
+		}
+		if len(ongoingCmds) > 0 &&
+			!utils.AskPrompt(fmt.Sprintf("found ongoing voyager commands: %s. Do you want to continue", strings.Join(ongoingCmds, ", "))) {
+			utils.ErrExit("aborting the end migration command")
+		}
+	} else {
+		log.Infof("no ongoing voyager commands found")
+	}
+
+	if bool(backupSchemaFiles) && !msr.ExportSchemaDone {
+		utils.PrintAndLog("backup schema files flag is set but schema export is not done, skipping schema backup...")
+		backupSchemaFiles = false
+	}
+
+	if backupDataFiles {
+		if !areOnDifferentFileSystems(exportDir, backupDir) {
+			return
+		}
+
+		if !msr.ExportDataDone {
+			utils.PrintAndLog("backup data files flag is set but data export is not done, skipping data backup...")
+			backupDataFiles = false
+			return
+		}
+
+		// verify that the size of backup-data dir to be greater the export-dir/data dir
+		exportDirDataSize, err := calculateDirSizeWithPattern(exportDir, "data/*.sql")
+		if err != nil {
+			utils.ErrExit("end migration: calculating export dir data size: %v", err)
+		}
+
+		backupDirSize, err := getFreeDiskSpace(backupDir)
+		if err != nil {
+			utils.ErrExit("end migration: calculating backup dir size: %v", err)
+		}
+
+		if exportDirDataSize >= int64(backupDirSize) {
+			utils.ErrExit(`end migration: backup directory free space is less than the export directory data size.
+			Please provide a backup directory with more free space than the export directory data size(%s).`, humanize.Bytes(uint64(exportDirDataSize)))
+		}
+	}
+}
+
+// NOTE: function is for Linux only (Windows won't work)
+// TODO: verify with dockerized voyager
+func areOnDifferentFileSystems(path1 string, path2 string) bool {
+	stat1 := syscall.Stat_t{}
+	stat2 := syscall.Stat_t{}
+
+	err1 := syscall.Stat(path1, &stat1)
+	err2 := syscall.Stat(path2, &stat2)
+
+	if err1 != nil || err2 != nil {
+		utils.ErrExit("end migration: getting file system info for %s and %s: %v, %v", path1, path2, err1, err2)
+	}
+
+	return stat1.Dev != stat2.Dev
+}
+
+func calculateDirSizeWithPattern(dirPath string, filePattern string) (int64, error) {
+	var size int64
+
+	fileMatches, err := filepath.Glob(filepath.Join(dirPath, filePattern))
+	if err != nil {
+		return 0, fmt.Errorf("matching the file pattern %q in the directory %q: %w", filePattern, dirPath, err)
+	}
+
+	for _, filePath := range fileMatches {
+		fileInfo, err := os.Stat(filePath)
+		if err != nil {
+			return 0, fmt.Errorf("getting file info for %q: %w", filePath, err)
+		}
+
+		if !fileInfo.IsDir() {
+			size += fileInfo.Size()
+		}
+	}
+
+	return size, nil
+}
+
+func getFreeDiskSpace(path string) (uint64, error) {
+	var stat syscall.Statfs_t
+	if err := syscall.Statfs(path, &stat); err != nil {
+		return 0, err
+	}
+	// calculate the free space in bytes
+	freeSpace := stat.Bfree * uint64(stat.Bsize)
+	return freeSpace, nil
+}
+
+func init() {
+	endCmd.AddCommand(endMigrationCmd)
```

Comment:
How are we receiving the various db passwords? They are not present in the msr!

Reply:
yeah it was because the `Uri` was present in conf. That has been fixed now in `main`.
Rebasing this branch with `main`